### ISSUE TITLE
Loss nan after add new loss

### ISSUE THREAD

**Issue by JoyHuYY1412** [facebookresearch/maskrcnn-benchmark#487]
## ❓ Questions and Help
I add some layers and a new loss in this network, however after several iterations, the features output from backbone is nan, and loss is nan then.

I have tried decrease the learning rate, add gradient clipping, but it seems the nan will still exists. Could you please give me some advice?

**Comment by fmassa** [comment]
I don't actually know about your loss. Having `nan` is tricky, and would require debugging the code and implementation.

I would try multiplying your new loss by 0 and seeing if you still have nans. If not, then find a scaling factor there that makes it stable maybe?

Apart from that, I don't have any other ideas ATM.

**Comment by JoyHuYY1412** [comment]
> I don't actually know about your loss. Having `nan` is tricky, and would require debugging the code and implementation.
> 
> I would try multiplying your new loss by 0 and seeing if you still have nans. If not, then find a scaling factor there that makes it stable maybe?
> 
> Apart from that, I don't have any other ideas ATM.

Thank you! As you pointed, the new loss is about 10 and old loss is about 1. So maybe the learning rate should be smaller and I should add a weight for new loss like 0.1. 
Am I right?

**Comment by fmassa** [comment]
Yes, that could potentially be a solution

**Comment by DeqiangWang** [comment]
@fmassa Hi, thanks your repository. I also meet loss nan. I have trained the model for 60000 iters(data:14000images, 8GPU, lr:0.002, batch_size=8),   and it seems convergence， but suddenly throw out nan.

2019-05-07 13:15:58,459 maskrcnn_benchmark.trainer INFO: eta: 18:33:17  iter: 529591  data: 0.2266 (0.2452)  loss_retina_cls: 0.1431 (0.2046)  time: 0.5795 (0.6050)  loss: 0.3987 (0.5962)  loss_retina_reg: 0.1508 (0.1981)  loss_mask: 0.1253 (0.1935)  lr: 0.002000  max mem: 4946
2019-05-07 13:15:59,020 maskrcnn_benchmark.trainer INFO: eta: 18:33:16  iter: 529592  data: 0.2266 (0.2452)  loss_retina_cls: 0.1431 (0.4451)  time: 0.5739 (0.6050)  loss: 0.3987 (1.2546)  loss_retina_reg: 0.1508 (0.4838)  loss_mask: 0.1253 (0.3258)  lr: 0.002000  max mem: 4946
2019-05-07 13:15:59,596 maskrcnn_benchmark.trainer INFO: eta: 18:33:15  iter: 529593  data: 0.2253 (0.2452)  loss_retina_cls: 0.1610 (nan)  time: 0.5739 (0.6050)  loss: 0.3987 (nan)  loss_retina_reg: 0.1641 (nan)  loss_mask: 0.1253 (nan)  lr: 0.002000  max mem: 4946
2019-05-07 13:16:00,140 maskrcnn_benchmark.trainer INFO: eta: 18:33:14  iter: 529594  data: 0.2253 (0.2452)  loss_retina_cls: 0.1610 (nan)  time: 0.5714 (0.6050)  loss: 0.3987 (nan)  loss_retina_reg: 0.1641 (nan)  loss_mask: 0.1253 (nan)  lr: 0.002000  max mem: 4946 

so, I thinks Iif we can set gradent clip to solve the issue?
